go/libraries/doltcore/env/actions/infer_schema.go

```diff
@@ -16,47 +16,71 @@ package actions
 
 import (
 	"context"
-	"errors"
 	"math"
 	"strconv"
 	"strings"
+	"time"
 
 	"github.com/google/uuid"
 
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/doltdb"
 	"github.com/liquidata-inc/dolt/go/libraries/doltcore/schema"
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/schema/typeinfo"
 	"github.com/liquidata-inc/dolt/go/libraries/doltcore/table"
 	"github.com/liquidata-inc/dolt/go/libraries/doltcore/table/pipeline"
+	"github.com/liquidata-inc/dolt/go/libraries/utils/funcitr"
 	"github.com/liquidata-inc/dolt/go/libraries/utils/set"
 	"github.com/liquidata-inc/dolt/go/store/types"
 )
 
 // StrMapper is a simple interface for mapping a string to another string
 type StrMapper interface {
 	// Map maps a string to another string.  If a string is not in the mapping ok will be false, otherwise it is true.
-	Map(str string) (mappedStr string, ok bool)
+	MaybeMap(str string) string
```

Comment:
Update the comment.  With the boolean return value removed, I don't know that I'd call this "MaybeMap" instead of just "Map"...  even if it just gets mapped to itself.